src/sagemaker/workflow/_utils.py

```diff
@@ -57,7 +60,7 @@ def __init__(
         description: str = None,
         source_dir: str = None,
         dependencies: List = None,
-        depends_on: Union[List[str], List[Step]] = None,
+        depends_on: Optional[List[Union[str, "Step", "StepCollection"]]] = None,
```

Comment:
The reason is the type annotations in `Step` and `StepCollection` can cause circular import errors since they're referencing each other.
The fix for this kind of error is to do a type checking import, which requires to add the "" quote for lower version of python
```
if TYPE_CHECKING:
    from sagemaker.workflow.steps import Step
```

Reply:
BTW, just found, the "Step" here does not need the quote. Good catch!

Reply:
Updated in the new commit